### PR TITLE
[ALLUXIO-2542] Enable customize top tier

### PR DESCRIPTION
**Pull request by apc999**
https://alluxio.atlassian.net/browse/ALLUXIO-2542

This PR aims to add support to customize the top tier dir freely in `alluxio-site.properties` or using `ALLUXIO_RAM_FOLDER`. In addition, it simplifies the logic 

Specifically:
- Calculating the default ram folder based on OS in Java land rather than in Bash
- Calculating the default worker memory size based on total physical memory in Java land rather than in Bash
- If top tier is not MEM, allows `NoMount`
- Don't hardcode default ram folder to system properties --- this permits setting the folder in site properties.

### CODE REVIEW

File: bin/alluxio-mount.sh [hunk 1]
```diff
@@ -224,18 +192,22 @@ ${DECL_MEM_SIZE_TO_BYTES}; ${DECL_MOUNT_LINUX}; mount_ramfs_linux $0"
...
+          ;;
+      esac
+      ;;
+    *)
+      echo -e ${USAGE} >&2
+      exit 1
+  esac
+}
+
+main "$@"
```
Comment (aaudiber) [inline bin/alluxio-mount.sh]: nit: newline

File: core/common/src/main/java/alluxio/Configuration.java [hunk 2]
```diff
@@ -95,6 +96,25 @@ public static void defaultInit() {
         String.valueOf(ChannelType.defaultType()));
     defaultProps.setProperty(PropertyKey.USER_NETWORK_NETTY_CHANNEL.toString(),
         String.valueOf(ChannelType.defaultType()));
+    // Set ramdisk volume according to OS type
```
Comment (aaudiber) [inline core/common/src/main/java/alluxio/Configuration.java]: this method is getting pretty long, do you think it makes sense to have a helper method for creating default properties?

File: bin/alluxio-mount.sh (hunk 1 above)
Comment (jsimsa) [inline bin/alluxio-mount.sh]: add newline

File: core/common/src/main/java/alluxio/Configuration.java [hunk 3]
```diff
@@ -95,6 +96,25 @@ public static void defaultInit() {
...
+    if (OSUtils.isLinux()) {
+      defaultProps
+          .setProperty(PropertyKey.WORKER_TIERED_STORE_LEVEL0_DIRS_PATH.toString(), "/mnt/ramdisk");
+    } else if (OSUtils.isMacOS()) {
+      defaultProps.setProperty(PropertyKey.WORKER_TIERED_STORE_LEVEL0_DIRS_PATH.toString(),
+          "/Volumes/ramdisk");
+    }
+    // Set a reasonable default for worker memory
+    try {
+      com.sun.management.OperatingSystemMXBean operatingSystemMXBean =
```
Comment (jsimsa) [inline core/common/src/main/java/alluxio/Configuration.java]: why not import this?

File: bin/alluxio-mount.sh (hunk 1 above)
Comment (apc999) [inline bin/alluxio-mount.sh]: done

Comment (apc999) [inline bin/alluxio-mount.sh]: done

File: core/common/src/main/java/alluxio/Configuration.java (hunk 2 above)
Comment (apc999) [inline core/common/src/main/java/alluxio/Configuration.java]: done

File: core/common/src/main/java/alluxio/Configuration.java (hunk 3 above)
Comment (apc999) [inline core/common/src/main/java/alluxio/Configuration.java]: done

**Comment by apc999** [comment]
@jsimsa @aaudiber Thanks for the comments. I addressed all of them. PTAL.

File: bin/alluxio-start.sh [hunk 4]
```diff
@@ -85,25 +83,26 @@ check_mount_mode() {
     Mount);;
     SudoMount);;
     NoMount)
-      is_ram_folder_mounted
+      local folder_type=$(${BIN}/alluxio getConf alluxio.worker.tieredstore.level0.alias)
```
Comment (jsimsa) [inline bin/alluxio-start.sh]: nit: call the variable `tier_alias`

File: bin/alluxio-start.sh [hunk 5]
```diff
@@ -85,25 +83,26 @@ check_mount_mode() {
...
+      local ram_folder=$(${BIN}/alluxio getConf alluxio.worker.tieredstore.level0.dirs.path)
```
Comment (jsimsa) [inline bin/alluxio-start.sh]: nit: call the variable `tier_path`

File: bin/alluxio-start.sh [hunk 6]
```diff
@@ -85,25 +83,26 @@ check_mount_mode() {
...
+      if [[ ${folder_type} != "MEM" ]]; then
```
Comment (gpang) [inline bin/alluxio-start.sh]: Is there any guarantee that the alias has to be called `MEM`? If so, is there any documentation saying that the ram must be the top tier, and it must be called `MEM`?

File: bin/alluxio-start.sh [hunk 7]
```diff
@@ -37,7 +37,8 @@ MOPT (Mount Option) is one of:
   SudoMount\tMount the configured RamFS using sudo.
            \tNotice: this will format the existing RamFS.
   NoMount  \tDo not mount the configured RamFS.
-           \tNotice: Use NoMount (Linux only) to use tmpFS to avoid sudo requirement.
+           \tNotice: to avoid sudo requirement but use tmpFS in Linux,
+             one can set ALLUXIO_RAM_FOLDER=/dev/shm and use NoMount.
```
Comment (gpang) [inline bin/alluxio-start.sh]: How is `ALLUXIO_RAM_FOLDER` expected to be used when you have remote workers? Will remote workers be able to use `/dev/shm`?

File: bin/alluxio-start.sh [hunk 8]
```diff
@@ -37,7 +37,8 @@ MOPT (Mount Option) is one of:
   SudoMount\tMount the configured RamFS using sudo.
            \tNotice: this will format the existing RamFS.
   NoMount  \tDo not mount the configured RamFS.
-           \tNotice: Use NoMount (Linux only) to use tmpFS to avoid sudo requirement.
+           \tNotice: to avoid sudo requirement but use tmpFS in Linux,
```
Comment (aaudiber) [inline bin/alluxio-start.sh]: `but use` -> `by using`

Comment (apc999) [inline bin/alluxio-start.sh]: done

File: bin/alluxio-start.sh (hunk 7 above)
Comment (apc999) [inline bin/alluxio-start.sh]: as talked offline, NoMount now has nothing to do with folder path, users need to set them separately.

File: bin/alluxio-start.sh (hunk 4 above)
Comment (apc999) [inline bin/alluxio-start.sh]: done

File: bin/alluxio-start.sh (hunk 6 above)
Comment (apc999) [inline bin/alluxio-start.sh]: no guarantee. if not called "MEM", no check enforced --- not known how to interpret the type any way.

**Comment by apc999** [comment]
@gpang @aaudiber  comments addressed

**Comment by jsimsa** [comment]
Jenkins, test this please

**Pull request merged**